File: app/src/test/java/com/kaciula/archiman/presentation/screen/home/HomePresenterTest.java
package com.kaciula.archiman.presentation.screen.home;


import com.kaciula.archiman.domain.usecases.GetUsers;
import com.kaciula.archiman.domain.util.SchedulerProvider;
import com.kaciula.archiman.domain.util.TrampolineSchedulerProvider;
import org.junit.Before;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class HomePresenterTest {

  @Mock HomeContract.View view;
  @Mock GetUsers getUsers;

  private HomePresenter presenter;

  @Before
  public void setupPresenter() {
    MockitoAnnotations.initMocks(this);

    SchedulerProvider schedulerProvider = new TrampolineSchedulerProvider();
    presenter = new HomePresenter(view, schedulerProvider, getUsers);
  }
}

File: app/src/test/java/com/kaciula/archiman/presentation/screen/home/HomePresenterTest.kt
package com.kaciula.archiman.presentation.screen.home


import com.kaciula.archiman.domain.usecases.GetUsers
import com.kaciula.archiman.domain.util.TrampolineSchedulerProvider
import org.junit.Before
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class HomePresenterTest {

    @Mock lateinit var view: HomeContract.View
    @Mock lateinit var getUsers: GetUsers

    private lateinit var presenter: HomePresenter

    @Before
    fun setupPresenter() {
        MockitoAnnotations.initMocks(this)

        val schedulerProvider = TrampolineSchedulerProvider()
        presenter = HomePresenter(view, schedulerProvider, getUsers)
    }
}
